pilot/pkg/security/authz/matcher/string_test.go

```diff
@@ -30,6 +32,7 @@ type testCase struct {
 }
 
 func TestStringMatcherWithPrefix(t *testing.T) {
+	features.EnableAuthzRegexMatching = true
```

Comment:
Do not set the flag to true for all tests, just set it to specific test case to make sure we cover both true and false cases.

Reply:
Done adding tests for both enabled and disabled case to StringMatcher and HeaderMatcher.